plugins/out_stackdriver/stackdriver.c

```diff
@@ -530,6 +530,38 @@ static int process_local_resource_id(const void *data, size_t bytes,
     return -1;
 }
 
+/*
+ * parse_labels
+ * - Iterate throught the original payload (obj) and find out the entry that matches
+ *   the labels_key 
+ * - Used to convert all labels under labels_key to root-level `labels` field
+ */
+static msgpack_object *parse_labels(struct flb_stackdriver *ctx, msgpack_object *obj)
+{
+    int i;
+    int len;
+    msgpack_object_kv *kv = NULL;
+
+    if (!obj || obj->type != MSGPACK_OBJECT_MAP) {
+        return NULL;
+    }
+
+    len = flb_sds_len(ctx->labels_key);
+    for (i = 0; i < obj->via.map.size; i++) {
+        kv = &obj->via.map.ptr[i];
+        if (flb_sds_casecmp(ctx->labels_key, kv->key.via.str.ptr, len) == 0) {
+            if (kv->val.type != MSGPACK_OBJECT_MAP) {
+                flb_plg_warn(ctx->ins, "the type of labels should be map");
+            }
+            return &kv->val;
```

Comment:
Does this mean only the first entry matching the label_key is returned?

Reply:
Yes. Only the first entry matching the label_key is returned. For the case that the payload already contains labels under Logging.googleapis/labels, we are going to use the **nest** plugin to handle it.

Reply:
Note: Add comments that it only matches the first entry with a matching key